internal/services/notification/webhooks.go

```diff
@@ -180,41 +179,34 @@ func (n *NotificationService) sendWebhooks(ctx context.Context, ev *rstypes.RunE
 		webhook.Run.Tasks[id] = task
 	}
 
-	if err := n.sendRunWebhook(ctx, webhook); err != nil {
-		return errors.WithStack(err)
-	}
+	var wh *types.RunWebhook
 
-	return nil
-}
+	err := n.d.Do(ctx, func(tx *sql.Tx) error {
+		payload, err := json.Marshal(webhook)
+		if err != nil {
+			return errors.WithStack(err)
+		}
 
-func (n *NotificationService) sendRunWebhook(ctx context.Context, webhook *RunWebhook) error {
-	c := &http.Client{}
+		wh = types.NewRunWebhook(tx)
+		wh.Payload = string(payload)
 
-	body, err := json.Marshal(webhook)
-	if err != nil {
-		return errors.WithStack(err)
-	}
+		if err := n.d.InsertOrUpdateRunWebhook(tx, wh); err != nil {
+			return errors.WithStack(err)
+		}
 
-	req, err := http.NewRequest("POST", n.c.WebhookURL, bytes.NewReader(body))
-	req = req.WithContext(ctx)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-	req.Header.Add("Content-Type", "application/json")
-	req.Header.Add(agolaEventHeader, string(AgolaEventRun))
+		runWebhookDelivery := types.NewRunWebhookDelivery(tx)
+		runWebhookDelivery.RunWebhookID = wh.ID
+		runWebhookDelivery.DeliveryStatus = types.DeliveryStatusNotDelivered
 
-	if n.c.WebhookSecret != "" {
-		sig256 := hmac.New(sha256.New, []byte(n.c.WebhookSecret))
-		_, err = io.MultiWriter(sig256).Write([]byte(body))
-		if err != nil {
+		if err := n.d.InsertOrUpdateRunWebhookDelivery(tx, runWebhookDelivery); err != nil {
```

Comment:
Isn't this just an insert?

Reply:
@sgotti I do not know what you mean

Reply:
Here you're inserting a new entry, not updating an existing one. So use InsertBla and not InsertOrUpdateBla.